modules/features2d/src/fast.cpp

```diff
@@ -435,11 +435,19 @@ void FAST(InputArray _img, std::vector<KeyPoint>& keypoints, int threshold, bool
     cv::Mat img = _img.getMat();
     CALL_HAL(fast_dense, hal_FAST, img, keypoints, threshold, nonmax_suppression, type);
 
-    size_t keypoints_count = 10000;
+    size_t keypoints_count = 1;
     keypoints.clear();
-    keypoints.resize(keypoints_count);
-    CALL_HAL(fast, cv_hal_FAST, img.data, img.step, img.cols, img.rows,
-             (uchar*)(keypoints.data()), &keypoints_count, threshold, nonmax_suppression, type);
+    uchar* kps = (uchar*)malloc(sizeof(KeyPoint) * keypoints_count);
```

Comment:
you don't need extra variable. Use the following code instead:

```
    KeyPoint* kps = (KeyPoint*)malloc(sizeof(kps[0]) * keypoints_count);
    int hal_ret = cv_hal_FAST(img.data, img.step, img.cols, img.rows, (void**)&kps,
                              &keypoints_count, threshold, nonmax_suppression, type, realloc);
    if (hal_ret == CV_HAL_ERROR_OK) {
        keypoints.assign(kps, kps + keypoints_count);
    }
    free(kps);
    if (hal_ret == CV_HAL_ERROR_OK) {
        return;
    }
```

Reply:
Done.